### PR TITLE
STY: pyupgrade to 3.9

### PR DESCRIPTION
**Pull request by mroeschke**

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -84,8 +84,6 @@
     from pandas import DataFrame
     from pandas.core.arrays import PeriodArray
 
-_midnight = time(0, 0)
```
Comment (rhshadrach) [inline pandas/core/arrays/datetimes.py:87]: We're certain this is no longer used?

File: pandas/tests/util/test_str_methods.py [hunk 2]
```diff
@@ -1,47 +1,3 @@
...
-        result = removesuffix(string, suffix)
-        assert result == expected
-
-else:
-    # NOTE: remove this file when pyupgrade --py39-plus removes
-    # the above block
-    pass
+# NOTE: remove this file when pyupgrade --py39-plus removes
+# the above block
+pass
```
Comment (rhshadrach) [inline pandas/tests/util/test_str_methods.py:3]: Remove this file?

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/tests/util/test_str_methods.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/util/test_str_methods.py:3]: Good catch!

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/arrays/datetimes.py:87]: Yeah I didn't find a reference to this variable anymore

```bash
% grep -R --include="*.py" _midnight pandas
pandas/io/formats/format.py:    not_midnight = values_int % one_day_nanos != 0  # type: ignore[operator]
pandas/io/formats/format.py:    both = np.logical_and(consider_values, not_midnight)  # type: ignore[arg-type]
pandas/tests/resample/test_datetime_index.py:def test_downsample_dst_at_midnight(unit):
pandas/tests/frame/methods/test_at_time.py:    def test_at_time_midnight(self, frame_or_series):
pandas/tests/indexes/datetimes/methods/test_shift.py:    def test_dti_shift_near_midnight(self, shift, result_time):
```

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @mroeschke

**Pull request merged**